### PR TITLE
Viper refactor

### PR DESCRIPTION
**Pull request by bcrochet**

### CODE REVIEW

File: cmd/preflight/cmd/check_container.go [hunk 1]
```diff
@@ -225,8 +227,7 @@ func checkContainerPositionalArgs(cmd *cobra.Command, args []string) error {
 
 // validateCertificationProjectID validates that the certification project id is in the proper format
 // and throws an error if the value provided is in a legacy format that is not usable to query pyxis
-func validateCertificationProjectID(cmd *cobra.Command, args []string) error {
-	viper := viper.Instance()
+func validateCertificationProjectID(viper *viper.Viper) error {
```
Comment (komish) [inline cmd/preflight/cmd/check_container.go:230]: Why do we need viper here? Just pass in the certification_project_id?

File: cmd/preflight/cmd/check_container_test.go [hunk 2]
```diff
@@ -7,49 +7,49 @@ import (
...
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
 )
 
 var _ = Describe("Check Container Command", func() {
 	BeforeEach(createAndCleanupDirForArtifactsAndLogs)
 
-	Context("when running the check container subcommand", func() {
-		Context("With all of the required parameters", func() {
+	When("running the check container subcommand", func() {
```
Comment (komish) [inline cmd/preflight/cmd/check_container_test.go:26]: WOAH, Woah, woah... unrelated change alert?! :laughing:

File: cmd/preflight/cmd/root_test.go [hunk 3]
```diff
@@ -85,17 +87,45 @@ var _ = Describe("cmd package utility functions", func() {
...
+					// This has to be written to current working directory in the memmapfs, since '.'
+					// as the config path is translated into an absolute path prior to the config
+					// file being read. So, here, we just create the file at that path, it just
+					// happens to be in a memmapfs
+					cwd, err := os.Getwd()
+					Expect(err).ToNot(HaveOccurred())
+					Expect(afero.WriteFile(fs, filepath.Join(cwd, "config.yaml"), bytes.NewBufferString(configFile).Bytes(), 0o644)).To(Succeed())
+				})
+				It("should read all of the config", func() {
+					initConfig(viper)
```
Comment (komish) [inline cmd/preflight/cmd/root_test.go:124]: while we're here, can we rename this? I would (and do currently) have no clue what this as I read it here in this context.

File: cmd/preflight/cmd/root_test.go [hunk 4]
```diff
@@ -8,12 +8,13 @@ import (
...
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/artifacts"
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/cli"
-	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/viper"
 
 	"github.com/go-logr/logr"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
+	spfviper "github.com/spf13/viper"
```
Comment (komish) [inline cmd/preflight/cmd/root_test.go:17]: No need for this alias, then?

File: cmd/preflight/cmd/check_container_test.go (hunk 2 above)
Comment (bcrochet) [reply cmd/preflight/cmd/check_container_test.go:26]: Is it though? :)

File: cmd/preflight/cmd/root_test.go (hunk 3 above)
Comment (bcrochet) [reply cmd/preflight/cmd/root_test.go]: Assume you meant the It() clause? Done.

File: cmd/preflight/cmd/root_test.go (hunk 4 above)
Comment (bcrochet) [reply cmd/preflight/cmd/root_test.go]: Nope.

**Review by skattoju — APPROVED** [review]
/lgtm

**Comment by komish** [comment]
Closing this due to age.
